tools/firtool/firtool.cpp

```diff
@@ -230,13 +235,20 @@ processBuffer(MLIRContext &context, TimingScope &ts, llvm::SourceMgr &sourceMgr,
         createCSEPass());
   }
 
+  if (inferMemories)
+    pm.nest<firrtl::CircuitOp>().nest<firrtl::FModuleOp>().addPass(
+        firrtl::createInferMemoriesPass());
+
   // Width inference creates canonicalization opportunities.
   if (inferWidths)
     pm.nest<firrtl::CircuitOp>().addPass(firrtl::createInferWidthsPass());
 
   if (inferResets)
     pm.nest<firrtl::CircuitOp>().addPass(firrtl::createInferResetsPass());
 
+  if (blackBoxMemory)
```

Comment:
Why move this?  Can you split any unrelated changes to a separate patch?

Reply:
I added a memory integration test that is making sure that this stuff was all working together, and I found that BlackBoxMemory isn't working anymore, and needs to run before LowerTypes.  I can move this + the integration test to a new PR.

Reply:
cool thanks